lib/util/cache/package/decorator.ts

```diff
@@ -42,6 +44,15 @@ interface CacheParameters {
   ttlMinutes?: number;
 }
 
+function getClassName(instance: unknown): string | undefined {
+  // istanbul ignore if: should never happen
+  if (!is.object(instance)) {
+    return undefined;
```

Comment:
we should throw if it's happening. undefined is a bad namespace.

or at least log a stack trace at debug level to see how it happens.